app/Services/Services/Sharing/ServiceOptionImporterService.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Pterodactyl - Panel
+ * Copyright (c) 2015 - 2017 Dane Everitt <dane@daneeveritt.com>.
+ *
+ * This software is licensed under the terms of the MIT license.
+ * https://opensource.org/licenses/MIT
+ */
+
+namespace Pterodactyl\Services\Services\Sharing;
+
+use Ramsey\Uuid\Uuid;
+use Illuminate\Http\UploadedFile;
+use Pterodactyl\Models\ServiceOption;
+use Illuminate\Database\ConnectionInterface;
+use Pterodactyl\Contracts\Repository\ServiceRepositoryInterface;
+use Pterodactyl\Exceptions\Service\Pack\InvalidFileUploadException;
+use Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface;
+use Pterodactyl\Contracts\Repository\ServiceVariableRepositoryInterface;
+use Pterodactyl\Exceptions\Service\ServiceOption\DuplicateOptionTagException;
+
+class ServiceOptionImporterService
+{
+    /**
+     * @var \Illuminate\Database\ConnectionInterface
+     */
+    protected $connection;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceRepositoryInterface
+     */
+    protected $serviceRepository;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceVariableRepositoryInterface
+     */
+    protected $serviceVariableRepository;
+
+    /**
+     * XMLImporterService constructor.
+     *
+     * @param \Illuminate\Database\ConnectionInterface                             $connection
+     * @param \Pterodactyl\Contracts\Repository\ServiceRepositoryInterface         $serviceRepository
+     * @param \Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface   $repository
+     * @param \Pterodactyl\Contracts\Repository\ServiceVariableRepositoryInterface $serviceVariableRepository
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        ServiceRepositoryInterface $serviceRepository,
+        ServiceOptionRepositoryInterface $repository,
+        ServiceVariableRepositoryInterface $serviceVariableRepository
+    ) {
+        $this->connection = $connection;
+        $this->repository = $repository;
+        $this->serviceRepository = $serviceRepository;
+        $this->serviceVariableRepository = $serviceVariableRepository;
+    }
+
+    /**
+     * Take an uploaded XML file and parse it into a new service option.
+     *
+     * @param \Illuminate\Http\UploadedFile $file
+     * @param int                           $service
+     * @return \Pterodactyl\Models\ServiceOption
+     *
+     * @throws \Pterodactyl\Exceptions\Model\DataValidationException
+     * @throws \Pterodactyl\Exceptions\Repository\RecordNotFoundException
+     * @throws \Pterodactyl\Exceptions\Service\Pack\InvalidFileUploadException
+     */
+    public function handle(UploadedFile $file, int $service): ServiceOption
```

Comment:
Should we just pass in the actual Service here instead. I'm not sure if the service option importer service should be in charge of finding the service too.

Reply:
This is how all of the other services are configured as well. Passing in the service model here would require already having that model and the required relations loaded. Doing it this way avoids that complexity. 

My entire goal with the repositories right now is to avoid doing SQL calls outside of them, so loading relations onto a model being passed in is not something I want to do.